applications/usb_uart_bridge/src/main.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <uart.h>
+#include <nrfx.h>
+#include <string.h>
+
+/* Overriding weak function to set iSerial runtime. */
+u8_t *usb_update_sn_string_descriptor(void)
+{
+	static u8_t buf[] = "PCA20035_12PLACEHLDRS";
+
+	snprintk(&buf[9], 13, "%04X%08X",
+		(uint32_t)(NRF_FICR->DEVICEADDR[1] & 0x0000FFFF)|0x0000C000,
+		(uint32_t)NRF_FICR->DEVICEADDR[0]);
+
+	return (u8_t *)&buf;
+}
+
+#define THREAD_STACKSIZE		1024
```

Comment:
Are you sure you need this much RAM for `power_thread` ?

Reply:
Clearly no. The thread should have no state and contains no nesting. What is a best practices way to estimate the maximum RAM needs of such a simple function?

Reply:
I reckon there is a Kconfig option to turn on logging the stack usage. Have never used it myself, but perhaps it's used in `nrf_desktop` or upstream. You could also proceed empirically :)

Reply:
The magic number seems to be 48 bytes. Done.